### PR TITLE
[editor] Fix "outdoor_seating=no is added to all POIs by the editor"

### LINKED ISSUE #9049: outdoor_seating=no is added to all POIs by the editor
Since the last release (2024-08-17-4-google) `outdoor_seating=no` is added to all POIs by the editor. The bug only occurs on android.

Example: https://www.openstreetmap.org/changeset/155576448 added `outdoor_seating=no` to a building

The bug looks limilar to #7642

### PR DESCRIPTION
**Pull request by Arthur-GYT**
All is in the title ^^.

Close #9049

### CODE REVIEW

File: android/app/src/main/java/app/organicmaps/editor/Editor.java [hunk 1]
```diff
@@ -89,9 +89,10 @@ public static void nativeSetBuildingLevels(String levels)
   public static native boolean nativeHasWifi();
   public static native void nativeSetHasWifi(boolean hasWifi);
 
-  public static void nativeSetSwitchInput(int id, Boolean switchValue, String checkedValue, String uncheckedValue)
+  public static void nativeSetSwitchInput(int id, Boolean switchValue, String checkedValue)
   {
-    nativeSetMetadata(id, switchValue ? checkedValue : uncheckedValue);
+    if(switchValue && nativeGetMetadata(id).isEmpty())
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/editor/Editor.java:94]:
```suggestion
    if (switchValue && nativeGetMetadata(id).isEmpty())
```

**Review by biodranik — COMMENTED** [review]
Will it work if user selects `No` explicitly? `No` value should be also saved if select explicitly by user (not by default)

**Comment by Arthur-GYT** [comment]
> Will it work if user selects `No` explicitly? `No` value should be also saved if select explicitly by user (not by default)

Not actually. Wouldn't we go for a system similar to self-service with just 2 options?

**Comment by biodranik** [comment]
self-service is implemented in the wrong way at the core and should be refactored, I mentioned it in one of the comments some days ago.

Can you please remind the current status of this feature? Does the generator filter out everything and leave only YES values, or does it store YES and NO?

**Comment by biodranik** [comment]
Then your code should do the following:
- Set YES if user selected YES
- Set NO if user selected NO
- Do not set anything if user did not explicitly specified YES or NO

**Comment by Arthur-GYT** [comment]
@biodranik It should be OK now. When the user interacts with the switch, it detects that there is a change.

**Comment by biodranik** [comment]
How other editable properties are implemented? Are there other similar to outdoor seating? Should it be a similar approach?

**Comment by map-per** [comment]
> How other editable properties are implemented?

I tested the implementation of 'Wifi', turns out that 'Wifi' doesn't work: #9126

**Comment by biodranik** [comment]
@map-per @Arthur-GYT and what about this PR? Does it have the same issue as https://github.com/organicmaps/organicmaps/issues/9126 ?

**Comment by map-per** [comment]
No, #9126 should not be a problem, because removing outdoor_seating is not possible with this implementation, one can only set yes or no

**Review by map-per — COMMENTED** [review]
I tested the PR on Android and it works as intended.

I don't like the approach of looking at `onCheckedChanged` though. We should rather use an approach with three options (no, unknown, yes) like in the iOS PR #9117

On the other hand it is also important to solve the problem in time for the next release, because it is not good that OM adds about 10,000 wrong outdoor_seating entries to OSM per month at the moment. So this PR could be a good quickfix.

**Comment by Arthur-GYT** [comment]
For this PR I've tested [Toogle button](https://m2.material.io/components/buttons#toggle-button) but it doesn't render :/

**Comment by Arthur-GYT** [comment]
For the code bellow, I have this render

```xml
<RelativeLayout
    android:id="@+id/block_outdoor_seating"
    style="@style/MwmWidget.Editor.MetadataBlock.Clickable">
  <ImageView
      style="@style/MwmWidget.Editor.MetadataIcon"
      app:srcCompat="@drawable/ic_outdoor_seating"
      tools:ignore="ContentDescription" />
  <TextView
      style="@style/MwmWidget.Editor.FieldLayout.EditText"
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      android:layout_centerVertical="true"
      android:layout_marginStart="@dimen/editor_margin_left"
      android:layout_toStartOf="@+id/sw__outdoor_seating"
      android:text="@string/outdoor_seating"/>
  <com.google.android.material.button.MaterialButtonToggleGroup
      android:id="@+id/sw__outdoor_seating"
      android:layout_width="wrap_content"
      android:layout_height="wrap_content"
      android:layout_alignParentEnd="true"
      android:layout_centerVertical="true"/>
    <Button
        android:id="@+id/button1"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="Button 1"
        style="?attr/materialButtonOutlinedStyle"
        />
    <Button
        android:id="@+id/button2"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="Button 2"
        style="?attr/materialButtonOutlinedStyle"
        />
  </com.google.android.material.button.MaterialButtonToggleGroup>
</RelativeLayout>
```

image

**Comment by biodranik** [comment]
We can undo the outdoor seating support if it is hard to fix it properly. Breaking OSM data is a blocker, and a hot fix is needed ASAP.

**Comment by biodranik** [comment]
@map-per @Arthur-GYT please write your detailed opinion on how we should tackle this important bug, and why, in the context of this PR and https://github.com/organicmaps/organicmaps/pull/9165

**Comment by Arthur-GYT** [comment]
Should be good now

Screenshot_20240831-122038

**Comment by Arthur-GYT** [comment]
Note: The tag is only deleted on the client side, not on OSM. I haven't found how to do it :/

**Comment by dvdmrtnz** [comment]
> Note: The tag is only deleted on the client side, not on OSM. I haven't found how to do it :/. Maybe a bug ??
> 
> EDIT: I think it's not possible yet to remove tags. I try with email input and it doesn't remove the tag on OSM.

* https://github.com/organicmaps/organicmaps/issues/6292

**Comment by Arthur-GYT** [comment]
I think it's better with icons.

And thanks @dvdmrtnz ^^

Screenshot_20240831-134733

**Comment by dvdmrtnz** [comment]
> I think it's better with icons.
> 
> And thanks @dvdmrtnz ^^
> 
> Screenshot_20240831-134733

Looking good! Could you switch them around so they match the iOS orientation? "X" on the left and "✓" on the right? Like in https://github.com/organicmaps/organicmaps/pull/7835#issuecomment-2053664755 and https://github.com/organicmaps/organicmaps/pull/9117

**Comment by Arthur-GYT** [comment]
> > I think it's better with icons.
> > And thanks @dvdmrtnz ^^
> > Screenshot_20240831-134733
> 
> Looking good! Could you switch them around so they match the iOS orientation? "X" on the left and "✓" on the right? Like in [#7835 (comment)](https://github.com/organicmaps/organicmaps/pull/7835#issuecomment-2053664755) and #9117

It's done ^^

Screenshot_20240831-140805

**Comment by biodranik** [comment]
What does "X" and "-" mean? Which one is "there is definitely no outdoor seating there" and "I don't know"?

**Comment by Arthur-GYT** [comment]
> What does "X" and "-" mean? Which one is "there is definitely no outdoor seating there" and "I don't know"?

- "X" -> there is definitely no outdoor seating there
- "-" -> I don't know

**Comment by biodranik** [comment]
How does this proposed cell look with a longer text translation in some other language, or with a very large system font size?

What was wrong with the text approach like self-service? With empty (or even better, "unknown") value for default state?

**Comment by Arthur-GYT** [comment]
Simply changing the font size will hide the text on the left.

Screenshot_20240831-153809

**Comment by Arthur-GYT** [comment]
And for self-service, we have this result
Screenshot_20240831-154004

**Comment by biodranik** [comment]
That doesn't look good. Implementing it as self-service would be better (but please see https://github.com/organicmaps/organicmaps/issues/9182 )

**Comment by Arthur-GYT** [comment]
> That doesn't look good. Implementing it as self-service would be better (but please see #9182 )

Even if I do a dialog, I need to use `editMapObject` and it reset the scroll animation.

**Comment by Arthur-GYT** [comment]
As far as I'm concerned, the current solutioncan be good enough for a corrective version.

**Review by map-per — CHANGES_REQUESTED** [review]
#### Review

- I would not add a new `RemoveMetadata` function to the editor, but encode the 'unknown' value with an empty string using `SetMetadata`. Then the behavior is similar to when users remove e.g. the instagram link.

- For the space problems in the layout I see two solutions:
   - Stick to the segmented button but move it to the next line (In a new line there would also be enough space to write out the options instead of using symbols. I prefer the 'no' '-' 'yes' approach from iOS, then there is less dis-ambiguity between 'x' and '-' (even in this PR discussion there were already questions about what 'x' and '-' mean))
   - Use a drop down menue (#9182). On the downside this is one click more for the users, but on the upside the meaning of the options is clearer as we can write out all options
   
   I'm indifferent between the two options.

#### Hot fix
For now I would use  #9165 for the hotfix to prevent further `outdoor_seating=no` additions to OSM.

**Comment by Arthur-GYT** [comment]
You're right, until we find a solution that satisfies everyone, it would be preferable to remove support for the tag while we merge this PR.

**Comment by map-per** [comment]
No, the bug was fixed in #9165

**Pull request closed without merging**